app/assets/javascripts/components/group/group_create_modal.es6.jsx

```diff
@@ -69,12 +55,25 @@ class GroupCreateModal extends CreateModal {
     };
   }
 
-  renderBody() {
-    var attributes = this.props.template.attributes;
-    var leaders = attributes['leaderships_attributes'];
-    if (this.props.template.errors) {
-      errors = this.props.template.errors;
+  generateValue(type) {
+    var groupables = this.props.groupables;
+    var leaderships = this.props.template.attributes['leaderships_attributes'];
+    if (type === 'primary' && leaderships[0]) {
+      var id = leaderships[0]['user_id'];
+      var user = groupables.find((groupable) => groupable.id == id);
```

Comment:
Turns out there's this cool `find` function, returns after first success so no need for `pop()`.